controllers/snapshot/snapshot_adapter.go

```diff
@@ -344,7 +344,7 @@ func (a *Adapter) EnsureSnapshotEnvironmentBindingExist() (reconciler.OperationR
 		return reconciler.RequeueWithError(err)
 	}
 
-	components, err := a.loader.GetAllApplicationComponents(a.client, a.context, a.application)
+	components, err := a.loader.GetAllSnapshotComponents(a.client, a.context, a.snapshot)
```

Comment:
You need to replace [ApplicationComponentsContextKey](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/snapshot/snapshot_adapter_test.go#L527) with the newly added  `SnapshotComponentsContextKey`.